tests/unit/service_test.py

```diff
@@ -150,6 +150,40 @@ def test_build_port_bindings_with_nonmatching_internal_ports(self):
         self.assertEqual(port_bindings["1000"], [("127.0.0.1", "1000")])
         self.assertEqual(port_bindings["2000"], [("127.0.0.1", "2000")])
 
+    def test_specifies_host_port_with_host_ip_and_port(self):
+        service = Service(
+            'foo',
+            image='foo',
+            ports=["127.0.0.1:1000:2000"])
+        self.assertEqual(service.specifies_host_port(), True)
```

Comment:
Rest of the service_test.py still uses `self.assertEqual`, I don't want to break the convention. What you're suggesting could go in as a separate PR that updates all tests IMO.